### DISCUSSION TITLE
Remote schema behind corporate proxy - possible ?

### DISCUSSION THREAD

**Discussion by oscar6echo** [hasura/graphql-engine#10242] [Question]
Adding a [remote schema](https://hasura.io/docs/latest/remote-schemas/adding-schema/) in hasura graphql engine is both very powerful and easy.  

But how to do it if the **remote graphql endpoint is behind a corporate proxy** ?

I could not find a way to do it if the hasura runs in a corporate env and the graphql API to add as a remote schema is outside i.e. behind the corporate proxy. Typically env variables `HTTP_PROXY`, `HTTPS_PROXY`, `NO_PROXY` are used to control proxy use. But I understand that hasura graphql engine does not support those.  

The only part of the [doc related to proxy](https://hasura.io/docs/latest/deployment/serve-behind-proxy/) is about how to expose hasura graphql endpoint behind a reverse proxy - a different topic.

Is it possible ?  
If so, how ?  
If not, that would be a useful feature :thinking:

**Comment by meetzaveri** [comment]
Hello @oscar6echo 

Thanks for raising this discussion. Reading your problem statement, it seems to me that if Hasura can be installed behind your corporate proxy, would that solve your issue ?

I would suggest you to look at the offerings of self-hosted or enterprise version of Hasura. You can find details of the enterprise pricing here in this [page](https://hasura.io/pricing/). 

In self-hosted version, you can run graphql-engine on your own behind your corporate infra. So that would eliminate the need for adding reverse proxy for your remote schema GraphQL endpoint. If you are using docker in your env, you can follow this getting-started [guide](https://hasura.io/docs/latest/getting-started/docker-simple/).

> Typically env variables HTTP_PROXY, HTTPS_PROXY, NO_PROXY are used to control proxy use. But I understand that hasura graphql engine does not support those.

I agree we don't have such customizations atm. But the simplest approach IMO is to run Hasura in your controlled environment.

**Comment by oscar6echo** [comment]
@meetzaveri thank you for the detailed answer.

But please note that I already self-host an open source hasura on corp premises, with corp auth - and it works superbely.   
The remote schema endpoint I mention is NOT under my control and lies OUTSIDE my own corp network, so only accessible through the corp proxy as it is a public resource.   

So what appears to be missing (unless I am mistaken) is a way to set proxy env vars applied to a remote schema.  
This limitation seems to be valid for open source and enterprise versions.  

Pls confirm / infirm / amend.

If confirrmed it would be a nice addition - and probably simple to implement.

**Reply by meetzaveri** [reply]
@oscar6echo Got it. I would like to know what kind of proxy env vars you are thinking that remote schema configuration should contain ? Feel free to create GH issue to have this tracked as a feature request.

**Reply by oscar6echo** [reply]
I think the usual env variables `HTTPS_PROX`, `HTTP_PROX` and equivalent  `WSS_PROX`, `WS_PROX` could be the **fields to add a remote schema configuration**. By default, they are not set and current behavior applies. If set then connections go through the proxy: `HTTPS_PROXY` for queries and mutations, and `WSS_PROXY` for subscriptions. The others are almost inusited now that all virtually all trafic is encrypted.

Cf. feature request https://github.com/hasura/graphql-engine/issues/10266
